src/peek_nth.rs

```diff
@@ -0,0 +1,121 @@
+use crate::size_hint;
+use std::collections::VecDeque;
+use std::iter::Fuse;
+use crate::PeekingNext;
+
+/// See [`peek_nth()`](../fn.peek_nth.html) for more information.
+#[derive(Clone, Debug)]
+pub struct PeekNth<I>
+where
+  I: Iterator,
+{
+  iter: Fuse<I>,
+  buf: VecDeque<I::Item>,
+}
+
+/// A drop-in replacement for `std::iter::Peekable` which adds a `peek_nth`
+/// method allowing the user to `peek` at a value several iterations forward
+/// without advancing the base iterator.
+///
+/// This differs from `multipeek` in that subsequent calls to `peek` or
+/// `peek_nth` will always return the same value until `next` is called
+/// (making `reset_peek` unnecessary).
+pub fn peek_nth<I>(iterable: I) -> PeekNth<I::IntoIter>
+where
+  I: IntoIterator,
+{
+  PeekNth {
+    iter: iterable.into_iter().fuse(),
+    buf: VecDeque::new(),
+  }
+}
+
+impl<I> PeekNth<I>
+where
+  I: Iterator,
+{
+  /// Works exactly like the `peek` method in `std::iter::Peekable`
+  pub fn peek(&mut self) -> Option<&I::Item> {
+    self.peek_nth(0)
+  }
+
+  /// Returns a reference to the `nth` value without advancing the iterator.
+  ///
+  /// # Examples
+  ///
+  /// Basic usage:
+  ///
+  /// ```rust
+  /// use itertools::peek_nth;
+  ///
+  /// let xs = vec![1,2,3];
+  /// let mut iter = peek_nth(xs.iter());
+  ///
+  /// assert_eq!(iter.peek_nth(0), Some(&&1));
+  /// assert_eq!(iter.next(), Some(&1));
+  ///
+  /// // The iterator does not advance even if we call `peek_nth` multiple times
+  /// assert_eq!(iter.peek_nth(0), Some(&&2));
+  /// assert_eq!(iter.peek_nth(1), Some(&&3));
+  /// assert_eq!(iter.next(), Some(&2));
+  ///
+  /// // Calling `peek_nth` past the end of the iterator will return `None`
+  /// assert_eq!(iter.peek_nth(1), None);
+  /// ```
+  pub fn peek_nth(&mut self, n: usize) -> Option<&I::Item> {
+    for _ in self.buf.len()..=n {
+      match self.iter.next() {
+        Some(next_item) => {
+          self.buf.push_back(next_item);
+        }
+        None => return None,
+      }
+    }
+
+    Some(&self.buf[n])
+  }
+}
+
+impl<I> Iterator for PeekNth<I>
+where
+  I: Iterator,
+{
+  type Item = I::Item;
+
+  fn next(&mut self) -> Option<I::Item> {
+    if self.buf.is_empty() {
+      self.iter.next()
+    } else {
+      self.buf.pop_front()
+    }
+  }
```

Comment:
The `is_empty` check isn't strictly needed, since `pop_front()` will return `None` if `buf` is empty. That check can be removed like this:
```suggestion
  fn next(&mut self) -> Option<I::Item> {
    self.buf.pop_front().or_else(|| self.iter.next())
  }
```